client/lib/reader-feed-subscriptions/index.js

```diff
@@ -146,8 +146,7 @@ var FeedSubscriptionStore = {
 		newSubscriptions = Immutable.List( subscriptionsWithState ); // eslint-disable-line new-cap
 
 		// Is it the last page?
-		if ( data.number < perPage ) {
-			// TODO: check to see if this is really the last page based on the count coming from the API
+		if ( data.number === 0 ) {
```

Comment:
> OK. So if we had an empty page for some reason... we may still stop early?

Yes, that could happen if all of the blogs were private, but it'd be very unlikely. We're generally asking for 50 results, and sometimes getting 48 or 49 back instead of 50.

We definitely should fix up the endpoint eventually, but this fixes the problem for the moment :)